### PR TITLE
Callbacks can short-circuit the Connection

### PR DESCRIPTION
**Pull request by ravster**
Changes from https://github.com/lostisland/hurley/pull/27

### CODE REVIEW

**Comment by rcruz** [comment]
+1

File: test/client_test.rb [hunk 1]
```diff
@@ -538,29 +574,59 @@ def test_builds_request
...
 
     def test_sets_before_callbacks
+      cb = Class.new
+      cb.class_eval do
+        def call(r)
+          5
+        end
+
+        def name
+          :fif
```
Comment (jsutlovic) [inline test/client_test.rb]: Does this name have a meaning?

Comment (ravster) [inline test/client_test.rb]: No, its just a name.  Changed to `:fifth` in line with the other callbacks.

**Comment by jsutlovic** [comment]
👍

**Pull request merged**